src/main/java/teammates/ui/controller/InstructorCourseEnrollSaveAction.java

```diff
@@ -44,19 +48,46 @@ public ActionResult execute() throws EntityDoesNotExistException {
 
         /* Process enrollment list and setup data for page result */
         try {
+            enrollErrorLines = new HashMap<>();
             List<StudentAttributes>[] students = enrollAndProcessResultForDisplay(studentsInfo, courseId);
             boolean hasSection = hasSections(students);
 
             InstructorCourseEnrollResultPageData pageData = new InstructorCourseEnrollResultPageData(account, sessionToken,
                                                                     courseId, students, hasSection, studentsInfo);
 
+            // TODO: Handle success status messages later
+
             statusToAdmin = "Students Enrolled in Course <span class=\"bold\">["
                             + courseId + "]:</span><br>" + sanitizedStudentsInfo.replace("\n", "<br>");
 
             return createShowPageResult(Const.ViewURIs.INSTRUCTOR_COURSE_ENROLL_RESULT, pageData);
 
         } catch (EnrollException e) {
+            InstructorCourseEnrollPageData pageData =
+                    new InstructorCourseEnrollPageData(account, sessionToken, courseId, studentsInfo);
+
+            List<String> exceptionMessages = new ArrayList<>(Arrays.asList(e.getMessage().split("<br>")));
```

Comment:
Concatenating data as string and then spliting is not the way to pass data around. Use some data structures like `ArrayList`